Library/Homebrew/test/test_cmd_audit.rb

```diff
@@ -13,53 +13,46 @@ def teardown
     FileUtils.rm_rf @dir
   end
 
-  def formula_text(name, text)
+  def formula_text(name, body = nil, patch = nil)
     path = Pathname.new "#{@dir}/#{name}.rb"
-    path.open("w") { |f| f.write text }
+    path.open("w") do |f|
+      f.write <<-EOS.undent
+        class #{Formulary.class_s(name)} < Formula
+          #{body}
+        end
+        #{patch}
+      EOS
+    end
     FormulaText.new path
   end
 
   def test_simple_valid_formula
-    ft = formula_text "valid", <<-EOS.undent
-      class Valid < Formula
-        url "http://www.example.com/valid-1.0.tar.gz"
-      end
-    EOS
+    ft = formula_text "valid", 'url "http://www.example.com/valid-1.0.tar.gz"'
 
     refute ft.has_DATA?, "The formula doesn't have DATA"
     refute ft.has_END?, "The formula doesn't have __END__"
-    assert ft.has_trailing_newline?, "The formula have a trailing newline"
+    assert ft.has_trailing_newline?, "The formula has a trailing newline"
 
-    assert ft =~ /\burl\b/, "The formula match 'url'"
+    assert ft =~ /\burl\b/, "The formula matches 'url'"
     assert_nil ft.line_number(/desc/), "The formula doesn't match 'desc'"
     assert_equal 2, ft.line_number(/\burl\b/)
+    assert ft.include?("Valid"), "The formula includes \"Valid\""
   end
 
   def test_trailing_newline
-    ft = formula_text "newline", "class Newline < Formula; end"
-    refute ft.has_trailing_newline?, "The formula doesn't have a trailing newline"
+    ft = formula_text "newline"
+    assert ft.has_trailing_newline?, "The formula doesn't have a trailing newline"
   end
 
   def test_has_data
-    ft = formula_text "data", <<-EOS.undent
-      class Data < Formula
-        patch :DATA
-      end
-    EOS
-
+    ft = formula_text "data", "patch :DATA"
     assert ft.has_DATA?, "The formula has DATA"
   end
 
   def test_has_end
-    ft = formula_text "end", <<-EOS.undent
-      class End < Formula
-      end
-      __END__
-      a patch here
-    EOS
-
+    ft = formula_text "end", "", "__END__\na patch here"
     assert ft.has_END?, "The formula has __END__"
-    assert_equal "class End < Formula\nend", ft.without_patch
+    assert_equal "class End < Formula\n  \nend", ft.without_patch
```

Comment:
Thanks. But I'm having trouble using `EOS.undent ` and getting the test to pass. 😞 

With it written this way:
```
  def test_has_end
    ft = formula_text "end", "", <<-EOS.undent
      __END__
      a patch here
    EOS
    assert ft.has_END?, "The formula must have __END__"
    assert_equal <<-EOS.undent, ft.without_patch
      class End < Formula
      end
    EOS
  end
```

The test returns this discrepancy:
```
 "class End < Formula
-end
-"
+  
+end"
```

I've tried adding newlines here and there, but with no luck so far.